### PR TITLE
Fix links in CONTRIBUTING.md, update ToC

### PR DESCRIPTION
**Pull request by BBlackwo**
Fixed a few broken links and updated the table of contents in the contributing guide

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -134,15 +135,17 @@ Enhancement suggestions are tracked as [GitHub issues](https://github.com/Start9
...
-- [`backend`] (backend/README.md) (Rust) is a command line utility, daemon, and software development kit that sets up and manages services and their environments, provides the interface for the ui, manages system state, and provides utilities for packaging services for EmbassyOS.
+- [`ui`](frontend/README.md) (Typescript Ionic Angular) is the code that is deployed to the browser to provide the user interface for EmbassyOS.
+- [`backend`](backend/README.md) (Rust) is a command line utility, daemon, and software development kit that sets up and manages services and their environments, provides the interface for the ui, manages system state, and provides utilities for packaging services for EmbassyOS.
 - `patch-db` - A diff based data store that is used to synchronize data between the front and backend.
-  - Notably, `patch-db` has a [client](patch-db/client/README.md) with its own dependency and installation requirements.
+  - Notably, `patch-db` has a [client](https://github.com/Start9Labs/patch-db/tree/master/client) with its own dependency and installation requirements.
```
Comment (ProofOfKeags) [inline CONTRIBUTING.md:143]: this points to master, and not the commit that is used in this repo (which may be master now but master is a dynamic pointer), you want to fix the relative link.

Comment (ProofOfKeags) [reply CONTRIBUTING.md:143]: upon further inspection, it seems that relative links don't work at all for submodules

**Review by ProofOfKeags — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -19,8 +19,9 @@ All types of contributions are encouraged and valued. See the [Table of Contents
 - [I Want To Contribute](#i-want-to-contribute)
   - [Reporting Bugs](#reporting-bugs)
   - [Suggesting Enhancements](#suggesting-enhancements)
+  - [Project Structure](#project-structure)
   - [Your First Code Contribution](#your-first-code-contribution)
-    - [Setting Up Your Development Environment](#setting-up-your-development-environment)
+    - [Setting up your development environment](#setting-up-your-development-environment)
```
Comment (kn0wmad) [inline CONTRIBUTING.md:24]: This copy change is now non-uniform to the rest of the options here.  They should all be capitalized or all not capitalized, but not a mixture

Comment (ProofOfKeags) [reply CONTRIBUTING.md:24]: oops, sorry for preemptive merge, would you like to be tagged as a reviewer on all doc changes to the repo @kn0wmad

Comment (BBlackwo) [reply CONTRIBUTING.md:24]: @kn0wmad nice spot. I fixed up the inconsistent casing in #1467.

**Pull request merged**